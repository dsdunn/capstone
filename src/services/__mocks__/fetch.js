export const getAllCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollectionsByCategory = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserInfo = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666
    }])
  })

export const getSearchResults = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      collections:[]
    }])
  })

export const postCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'
    }])
  })