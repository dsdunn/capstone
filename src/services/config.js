export default {
  apiKey: '66642069',
  url: 'https://collecshare.herokuapp.com/api/v1'
}