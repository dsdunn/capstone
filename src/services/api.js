import config from './config';

export const apiFetch = (path, method = 'GET', payload = null) => {
  let body = payload ? JSON.stringify(payload) : null
  return fetch(`${config.url}${path}`, {
    method,
    headers: {
     'Content-Type': 'application/json',
     'Accept': 'application/json'
    },
    body
  })
}

export const apiFormDataFetch = (path, method = 'PUT', payload = null) => {
  console.log(path, method, payload);
  let body = payload; 
  return fetch(`${config.url}${path}`, {
    method,
    body
  })
}
