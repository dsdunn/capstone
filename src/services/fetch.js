import { apiFetch, apiFormDataFetch } from './api'

export const postUserInfo = user => {
  return apiFetch('/users', 'POST', user)
}

export const getUserInfo = uid => {
  return apiFetch(`/users/${uid}`, 'GET')
  .then(response => response.json())
}

export const putUserInfo = (userInfo, uid) => {
  return apiFormDataFetch(`/users/${uid}`, 'PUT', userInfo)
  .then(response => console.log(response))
  .then(response => response.json())
}

export const postCollection = collection => {
  return apiFetch(`/collections`, `POST`, collection)
  .then(response => response.json())
}

export const getAllCollections = () => {
  return apiFetch('/collections')
  .then(response => response.json())
}

export const getUserCollections = uid => {
  return apiFetch(`/collections?uid=${uid}`)
  .then(response => response.json())
}