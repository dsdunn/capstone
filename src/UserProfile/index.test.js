import React from 'react';
import { shallow } from 'enzyme';
import { UserProfile, mapStateToProps } from './';
import { getUserCollections } from '../services/fetch'

jest.mock('../services/fetch');

describe('UserProfile', () => {
  let wrapper;
  let mockProfile;

  beforeEach(() => {
    mockProfile = {
      avatar: '2 comes out in 2020',
      username: 'username? username for what?',
      location: 'somewhere else',
      bio: 'dome'
    }
    wrapper = shallow(<UserProfile profile={mockProfile} /> )
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should display a message if no collections in state', () => {
    wrapper.setState({ collections: [] })
    expect(wrapper.find('.message').length).toEqual(1);
  })

  it('should return a props object', () => {
    const mockState = {profile: {}};
    const expected = {"profile": {}};

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})
