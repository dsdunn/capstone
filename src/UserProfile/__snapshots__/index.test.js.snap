// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile
    profile={
        Object {
            "avatar": "2 comes out in 2020",
            "bio": "dome",
            "location": "somewhere else",
            "username": "username? username for what?",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-profile-background"
/>,
        <div
          className="user-profile"
>
          <h1
                    className="profile-display-name"
          >
                    username? username for what?
          </h1>
          <section
                    className="profile-top"
          >
                    <div>
                              <p
                                        className="profile-location"
                              >
                                        LOCATION: 
                                        <span
                                                  className="profile-location-span"
                                        >
                                                  somewhere else
                                        </span>
                              </p>
                              <p
                                        className="profile-bio"
                              >
                                        ABOUT ME: 
                                        <span
                                                  className="profile-bio-span"
                                        >
                                                  dome
                                        </span>
                              </p>
                    </div>
                    <div
                              className="profile-avatar-container"
                    >
                              <img
                                        alt="profile"
                                        className="profile-img"
                                        src="https://collecshare.herokuapp.com/2 comes out in 2020"
                              />
                    </div>
          </section>
          <section
                    className="profile-collections"
          >
                    <p
                              className="message"
                    >
                              No Collections yet!
                    </p>
          </section>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "user-profile-background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="profile-display-name"
>
              username? username for what?
</h1>,
            <section
              className="profile-top"
>
              <div>
                            <p
                                          className="profile-location"
                            >
                                          LOCATION: 
                                          <span
                                                        className="profile-location-span"
                                          >
                                                        somewhere else
                                          </span>
                            </p>
                            <p
                                          className="profile-bio"
                            >
                                          ABOUT ME: 
                                          <span
                                                        className="profile-bio-span"
                                          >
                                                        dome
                                          </span>
                            </p>
              </div>
              <div
                            className="profile-avatar-container"
              >
                            <img
                                          alt="profile"
                                          className="profile-img"
                                          src="https://collecshare.herokuapp.com/2 comes out in 2020"
                            />
              </div>
</section>,
            <section
              className="profile-collections"
>
              <p
                            className="message"
              >
                            No Collections yet!
              </p>
</section>,
          ],
          "className": "user-profile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "username? username for what?",
              "className": "profile-display-name",
            },
            "ref": null,
            "rendered": "username? username for what?",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <p
                                    className="profile-location"
                  >
                                    LOCATION: 
                                    <span
                                                      className="profile-location-span"
                                    >
                                                      somewhere else
                                    </span>
                  </p>
                  <p
                                    className="profile-bio"
                  >
                                    ABOUT ME: 
                                    <span
                                                      className="profile-bio-span"
                                    >
                                                      dome
                                    </span>
                  </p>
</div>,
                <div
                  className="profile-avatar-container"
>
                  <img
                                    alt="profile"
                                    className="profile-img"
                                    src="https://collecshare.herokuapp.com/2 comes out in 2020"
                  />
</div>,
              ],
              "className": "profile-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="profile-location"
>
                      LOCATION: 
                      <span
                                            className="profile-location-span"
                      >
                                            somewhere else
                      </span>
</p>,
                    <p
                      className="profile-bio"
>
                      ABOUT ME: 
                      <span
                                            className="profile-bio-span"
                      >
                                            dome
                      </span>
</p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "LOCATION: ",
                        <span
                          className="profile-location-span"
>
                          somewhere else
</span>,
                      ],
                      "className": "profile-location",
                    },
                    "ref": null,
                    "rendered": Array [
                      "LOCATION: ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "somewhere else",
                          "className": "profile-location-span",
                        },
                        "ref": null,
                        "rendered": "somewhere else",
                        "type": "span",
                      },
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "ABOUT ME: ",
                        <span
                          className="profile-bio-span"
>
                          dome
</span>,
                      ],
                      "className": "profile-bio",
                    },
                    "ref": null,
                    "rendered": Array [
                      "ABOUT ME: ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "dome",
                          "className": "profile-bio-span",
                        },
                        "ref": null,
                        "rendered": "dome",
                        "type": "span",
                      },
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="profile"
                    className="profile-img"
                    src="https://collecshare.herokuapp.com/2 comes out in 2020"
/>,
                  "className": "profile-avatar-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "profile",
                    "className": "profile-img",
                    "src": "https://collecshare.herokuapp.com/2 comes out in 2020",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="message"
>
                No Collections yet!
</p>,
              "className": "profile-collections",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No Collections yet!",
                "className": "message",
              },
              "ref": null,
              "rendered": "No Collections yet!",
              "type": "p",
            },
            "type": "section",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-profile-background"
/>,
          <div
            className="user-profile"
>
            <h1
                        className="profile-display-name"
            >
                        username? username for what?
            </h1>
            <section
                        className="profile-top"
            >
                        <div>
                                    <p
                                                className="profile-location"
                                    >
                                                LOCATION: 
                                                <span
                                                            className="profile-location-span"
                                                >
                                                            somewhere else
                                                </span>
                                    </p>
                                    <p
                                                className="profile-bio"
                                    >
                                                ABOUT ME: 
                                                <span
                                                            className="profile-bio-span"
                                                >
                                                            dome
                                                </span>
                                    </p>
                        </div>
                        <div
                                    className="profile-avatar-container"
                        >
                                    <img
                                                alt="profile"
                                                className="profile-img"
                                                src="https://collecshare.herokuapp.com/2 comes out in 2020"
                                    />
                        </div>
            </section>
            <section
                        className="profile-collections"
            >
                        <p
                                    className="message"
                        >
                                    No Collections yet!
                        </p>
            </section>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "user-profile-background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="profile-display-name"
>
                username? username for what?
</h1>,
              <section
                className="profile-top"
>
                <div>
                                <p
                                                className="profile-location"
                                >
                                                LOCATION: 
                                                <span
                                                                className="profile-location-span"
                                                >
                                                                somewhere else
                                                </span>
                                </p>
                                <p
                                                className="profile-bio"
                                >
                                                ABOUT ME: 
                                                <span
                                                                className="profile-bio-span"
                                                >
                                                                dome
                                                </span>
                                </p>
                </div>
                <div
                                className="profile-avatar-container"
                >
                                <img
                                                alt="profile"
                                                className="profile-img"
                                                src="https://collecshare.herokuapp.com/2 comes out in 2020"
                                />
                </div>
</section>,
              <section
                className="profile-collections"
>
                <p
                                className="message"
                >
                                No Collections yet!
                </p>
</section>,
            ],
            "className": "user-profile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "username? username for what?",
                "className": "profile-display-name",
              },
              "ref": null,
              "rendered": "username? username for what?",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <p
                                        className="profile-location"
                    >
                                        LOCATION: 
                                        <span
                                                            className="profile-location-span"
                                        >
                                                            somewhere else
                                        </span>
                    </p>
                    <p
                                        className="profile-bio"
                    >
                                        ABOUT ME: 
                                        <span
                                                            className="profile-bio-span"
                                        >
                                                            dome
                                        </span>
                    </p>
</div>,
                  <div
                    className="profile-avatar-container"
>
                    <img
                                        alt="profile"
                                        className="profile-img"
                                        src="https://collecshare.herokuapp.com/2 comes out in 2020"
                    />
</div>,
                ],
                "className": "profile-top",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="profile-location"
>
                        LOCATION: 
                        <span
                                                className="profile-location-span"
                        >
                                                somewhere else
                        </span>
</p>,
                      <p
                        className="profile-bio"
>
                        ABOUT ME: 
                        <span
                                                className="profile-bio-span"
                        >
                                                dome
                        </span>
</p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "LOCATION: ",
                          <span
                            className="profile-location-span"
>
                            somewhere else
</span>,
                        ],
                        "className": "profile-location",
                      },
                      "ref": null,
                      "rendered": Array [
                        "LOCATION: ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "somewhere else",
                            "className": "profile-location-span",
                          },
                          "ref": null,
                          "rendered": "somewhere else",
                          "type": "span",
                        },
                      ],
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "ABOUT ME: ",
                          <span
                            className="profile-bio-span"
>
                            dome
</span>,
                        ],
                        "className": "profile-bio",
                      },
                      "ref": null,
                      "rendered": Array [
                        "ABOUT ME: ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "dome",
                            "className": "profile-bio-span",
                          },
                          "ref": null,
                          "rendered": "dome",
                          "type": "span",
                        },
                      ],
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="profile"
                      className="profile-img"
                      src="https://collecshare.herokuapp.com/2 comes out in 2020"
/>,
                    "className": "profile-avatar-container",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "profile",
                      "className": "profile-img",
                      "src": "https://collecshare.herokuapp.com/2 comes out in 2020",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="message"
>
                  No Collections yet!
</p>,
                "className": "profile-collections",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No Collections yet!",
                  "className": "message",
                },
                "ref": null,
                "rendered": "No Collections yet!",
                "type": "p",
              },
              "type": "section",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
